### PR TITLE
TF-TRT Conv2d op conversion dynamic shape mode

### PR DESCRIPTION
**Pull request by tfeher**
This PR improves the Conv2d op converter to handle explicit batch and dynamic shape input data.
Tagging @bixia1 for review.

### CODE REVIEW

**Comment by gbaned** [comment]
@tfeher Can you please resolve conflicts? Thanks!

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 1]
```diff
@@ -2157,11 +2163,14 @@ Status ConvertConv2DHelper(OpConverterParams* params, int group,
   }
   // Dimensions of transposed tensor.
   const auto tensor_dim = tensor->getDimensions();
+  const int tensor_c = tensor_dim.d[params->use_implicit_batch ? 0 : 1];
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc:2166]: Shall we rename tensor_c to c_dim_size? Tensor_h/w are not used.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 2]
```diff
@@ -4037,15 +4037,16 @@ TEST_F(OpConverterTest, ConvertSlice) {
   }
 }
 
-TEST_F(OpConverterTest, ConvertConv2D) {
+TEST_P(OpConverterTest1, ConvertConv2D) {
   // Get nodedef for Conv2D layer.
+  DataType dtype = tf_dtype;
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:4042]: I am thinking to use tf_type and trt_type consistently in the TRT code. Shall we do this to the new code and make change to the existing code when we touch them for other reason?

**Review by tfeher — COMMENTED** [review]
Thanks Bixia for the review, I have addressed the issues.

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 2 above)
Comment (bixia1) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: Yes, Let's try to use tf_type consistently.

**Review by bixia1 — APPROVED** [review]

**Pull request merged**